react/src/tests/PersonalData-test.js

```diff
@@ -3,16 +3,460 @@ import React from 'react';
+
+    expect(form.hasClass('form-horizontal')).toBeTruthy();
+    expect(form.contains(fieldset.get(0))).toBeTruthy();
+    expect(fieldset.hasClass('tabpane')).toBeTruthy();
+    expect(fieldset.contains(language.get(0))).toBeTruthy();
+    expect(fieldset.contains(surname.get(0))).toBeTruthy();
+    expect(fieldset.contains(given_name.get(0))).toBeTruthy();
+    expect(fieldset.contains(display_name.get(0))).toBeTruthy();
+
+    expect(form.props()).toContain({role: 'form'});
+    expect(fieldset.props()).toContain({id: 'personal-data-form'});
+
+    expect(props.handleSave.calls.length).toEqual(0);
+    button.props().onClick();
+    expect(props.handleSave.calls.length).toEqual(1);
+
+  });
+
+});
+
+
+import { Provider } from 'react-redux';
+import { IntlProvider, addLocaleData } from 'react-intl';
+import PersonalDataContainer from "containers/PersonalData";
+
+const messages = require('../../i18n/l10n/en');
+addLocaleData('react-intl/locale-data/en');
+
+const fakeStore = (state) => ({
+  default: () => {},
+  dispatch: createSpy(),
+  subscribe: createSpy(),
+  getState: () => ({ ...state })
+});
+
+describe("PersonalData Container", () => {
+  let fulltext,
+    given_name,
+    fulldom,
+    surname,
+    display_name,
+    language,
+    mockProps,
+    wrapper,
+    dispatch;
+
+  beforeEach(() => {
+    const store = fakeStore({
+      personal_data: {
+        is_fetching: false,
+        failed: false,
+        given_name: '',
+        surname: '',
+        display_name: '',
+        language: '',
+      },
+      config: {PERSONAL_DATA_URL: 'http://localhost/services/personal-data/user'},
+    });
+
+    mockProps = {
+        given_name: 'Pablo',
+        surname: 'Iglesias',
+        display_name: 'Pablo',
+        language: 'en'
+    };
+
+
+    wrapper = mount(
+        <IntlProvider locale={'en'} messages={messages}>
+          <Provider store={store}>
+            <PersonalDataContainer {...mockProps}/>
+          </Provider>
+        </IntlProvider>
+    );
+    fulldom = wrapper.find(PersonalDataContainer);
+    fulltext = wrapper.find(PersonalDataContainer).text();
+    given_name = wrapper.find(PersonalDataContainer).props().given_name;
+    surname = wrapper.find(PersonalDataContainer).props().surname;
+    display_name = wrapper.find(PersonalDataContainer).props().display_name;
+    language = wrapper.find(PersonalDataContainer).props().language;
+    dispatch = store.dispatch;
   });
-  
-  it("should be equal", function (){
-    assert.equal(2, 2);
+
+
+  afterEach(() => {
+    fetchMock.restore()
+  });
+
+  it("Renders", () => {
+      expect(language).toEqual('en');
+      expect(given_name).toEqual('Pablo');
+      expect(surname).toEqual('Iglesias');
+      expect(display_name).toEqual('Pablo');
+  });
+
+  it("Clicks", () => {
+
+    fetchMock.post('http://localhost/services/personal-data/user',
+       {
+        type: actions.POST_USERDATA_SUCCESS
+      });
```

Comment:
In these tests checking for UI interactions, we  should be able to add a payload to the `fetchMock`ed requests, and then check that the data in the payload is in the state and in the UI.